pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/ExtensibleLoadManagerImpl.java

```diff
@@ -982,17 +1017,28 @@ protected void monitor() {
     }
 
     public void disableBroker() throws Exception {
+        // TopicDoesNotExistException might be thrown and it's not recoverable. Enable this flag to exit playFollower()
+        // or playLeader() quickly.
+        if (!state.compareAndSet(State.RUNNING, State.DISABLED)) {
+            failForUnexpectedState("disableBroker");
+        }
+        stopLoadDataReportTasks();
         serviceUnitStateChannel.cleanOwnerships();
-        leaderElectionService.close();
         brokerRegistry.unregister();
+        leaderElectionService.close();
+        final var availableBrokers = brokerRegistry.getAvailableBrokersAsync()
+                .get(conf.getMetadataStoreOperationTimeoutSeconds(), TimeUnit.SECONDS);
+        if (availableBrokers.isEmpty()) {
+            close();
+        }
         // Close the internal topics (if owned any) after giving up the possible leader role,
         // so that the subsequent lookups could hit the next leader.
         closeInternalTopics();
     }
 
     private void closeInternalTopics() {
         List<CompletableFuture<Void>> futures = new ArrayList<>();
-        for (String name : INTERNAL_TOPICS) {
+        for (String name : Set.of(BROKER_LOAD_DATA_STORE_TOPIC, TOP_BUNDLES_LOAD_DATA_STORE_TOPIC)) {
```

Comment:
Could you explain why this change is required?

Reply:
See the issue 5 that I added in the PR description.

If the `loadbalancer-service-unit-state` topic's owner happens to be the current broker, the producers could be disconnected. Then in `unloadNamespaceBundleAsync` it would fail to send unload related events.

Reply:
It's not required. I reverted this change first.